src/commands/createFunction/durableSteps/sql/SqlServerNameStep.ts

```diff
@@ -0,0 +1,48 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { SqlManagementClient } from '@azure/arm-sql';
+import { delay } from '@azure/ms-rest-js';
```

Comment:
Same as #3390, this is going to have to be removed for the web work, so we should avoid it now. It seems that we need a debounce function to use along with `checkNameAvailability`, this should probably go into `@microsoft/vscode-azext-utils`.

The current approach also isn't actually reducing the number of web calls, just delaying when they occur. A properly-implemented debounce function could do that. See https://github.com/microsoft/compose-language-service/blob/main/src/service/utils/debounce.ts as an example; this approach _cancels_ any still-to-be-executed calls when a new one comes in--this way, nothing executes until the user has stopped typing for X milliseconds.